### ISSUE TITLE
SubscriptionRegistrationResolve never called in 1.10.0 since field instrumenter is gone

### ISSUE THREAD

**Issue by gastonmorixe** [rmosolgo/graphql-ruby#2692]
This makes `events` always empty. 

I solved it somehow like this, Would love to know if this is ok or how should it be? it's really hard to follow the code since the gem is huge. 

Came around to this because `write_subscription` was never being called. I was trying AnyCable.

https://github.com/Envek/graphql-anycable/blob/master/lib/graphql/subscriptions/anycable_subscriptions.rb#L87

Thank you

```ruby

      # If needed, prepare to gather events which this query subscribes to
      def before_query(query)
        
        if query.subscription? && !query.subscription_update?
          query.context.namespace(:subscriptions)[:events] = []

          ctx = query.context
          events = ctx.namespace(:subscriptions)[:events]
          query_node_field = query.selected_operation.selections.first
          event_name = query_node_field.name
          field = query.get_field(@schema.subscription, event_name)
          args = field.arguments
          events << Subscriptions::Event.new(
            field: field,
            name: field.name,
            arguments: args,
            context: ctx,
          )
        end
      end
```

**Comment by Envek** [comment]
Most probably we need to reimplement that field instrumenter on our side :thinking:

**Comment by gastonmorixe** [comment]
Little better maybe, works for ActionCable too

```ruby
module GraphQL
  class Subscriptions
    class CustomInstrumentation < GraphQL::Subscriptions::Instrumentation
      def before_query(query)
        if query.subscription? && !query.subscription_update?
          events = []
          ctx = query.context
          query_node_field = query.selected_operation.selections.each do |query_node_field|
            event_name = query_node_field.name
            field = query.get_field(@schema.subscription, event_name) # i.e. 'userCreated'
            events << Subscriptions::Event.new(
              field: field,
              name: field.name,
              arguments: field.arguments,
              context: ctx
            )
          end
          ctx.namespace(:subscriptions)[:events] = events
        end
      end
    end
    class CustomSubscriptions < ActionCableSubscriptions # GraphQL::Subscriptions
      def self.use(defn, options = {})
        schema = defn.is_a?(Class) ? defn : defn.target

        if schema.subscriptions
          raise ArgumentError, "Can't reinstall subscriptions. #{schema} is using #{schema.subscriptions}, can't also add #{self}"
        end

        instrumentation = Subscriptions::CustomInstrumentation.new(schema: schema)
        defn.instrument(:query, instrumentation)
        # defn.instrument(:field, instrumentation)
        options[:schema] = schema
        schema.subscriptions = new(**options)
        nil
      end
    end
  end
end
```

**Comment by modosc** [comment]
i think we're seeing this in `1.9.18` when using the new interpreter and class-based subscriptions. when i add `binding.pry` [here](https://github.com/rmosolgo/graphql-ruby/blob/1.9.x/lib/graphql/subscriptions/instrumentation.rb#L32) `events` is always `nil` and subscriptions never fire off. it's weird too - in `MySubscriptionClass#update`, `object` is definitely populated and returned, but the actual subscription response is always empty.

**Comment by rmosolgo** [comment]
It sounds like this is an issue with anycable subscriptions, right? I'll close this out -- please let me know if I've misunderstood and there's something required from GraphQL-Ruby here!
